app/src/main/java/com/kickstarter/ui/viewholders/ProjectCardViewHolder.kt

```diff
@@ -0,0 +1,291 @@
+        viewModel.outputs.metadataViewGroupBackgroundDrawable()
+            .compose(bindToLifecycle())
+            .compose(Transformers.observeForUI())
+            .subscribe { binding.projectMetadataView.projectMetadataViewGroup.background = ContextCompat.getDrawable(context(), it) }
+        viewModel.outputs.metadataViewGroupIsGone()
+            .compose(bindToLifecycle())
+            .compose(Transformers.observeForUI())
+            .subscribe(ViewUtils.setGone(binding.projectMetadataView.projectMetadataViewGroup))
+        viewModel.outputs.savedViewGroupIsGone()
+            .compose(bindToLifecycle())
+            .compose(Transformers.observeForUI())
+            .subscribe(ViewUtils.setGone(binding.projectMetadataView.savedViewGroup))
+        viewModel.outputs.fundingSuccessfulViewGroupIsGone()
+            .compose(bindToLifecycle())
+            .compose(Transformers.observeForUI())
+            .subscribe(ViewUtils.setGone(binding.projectStateViewGroup.fundingSuccessfulViewGroup))
+        viewModel.outputs.setDefaultTopPadding()
+            .compose(bindToLifecycle())
+            .compose(Transformers.observeForUI())
+            .subscribe { setDefaultTopPadding(it) }
+    }
+
+    @Throws(Exception::class)
+    override fun bindData(data: Any?) {
+        val projectAndParams = ObjectUtils.requireNonNull(data as Pair<Project, DiscoveryParams>?)
+        viewModel.inputs.configureWith(projectAndParams)
+    }
+
+    private fun setSubcategoryTextView(subcategory: String) {
+        binding.projectCardTags.subcategoryTextView.text = subcategory
+    }
+
+    private fun setStyledNameAndBlurb(nameAndBlurb: Pair<String, String>) {
+        val nameString = if (ProjectUtils.isProjectNamePunctuated(nameAndBlurb.first)) nameAndBlurb.first.toString() + " " else nameAndBlurb.first.toString() + ": "
+        val blurbString = nameAndBlurb.second
+        val styledString = SpannableString(nameString + blurbString)
+        styledString.setSpan(ForegroundColorSpan(context().getColor(R.color.kds_support_700)), 0, nameString.length, 0)
+        styledString.setSpan(
+            ForegroundColorSpan(context().getColor(R.color.kds_support_400)),
+            nameString.length,
+            nameString.length + blurbString.length,
+            0
+        )
+        binding.nameAndBlurbTextView.text = styledString
+    }
+
+    private fun resizeProjectImage(avatarUrl: String?) {
+        val targetImageWidth = (ViewUtils.getScreenWidthDp(context()) * ViewUtils.getScreenDensity(context()) - context().resources.getDimension(R.dimen.grid_4)).toInt()
+        val targetImageHeight = ProjectUtils.photoHeightFromWidthRatio(targetImageWidth)
+
+        binding.projectCardPhoto.photo.maxHeight = targetImageHeight
+
+        Picasso.with(context())
+            .load(avatarUrl)
```

Comment:
since `avatarUrl` is of type `String?` we should also be checking here if avatarUrl is null since this will cause a crash if that is the case.